app/src/modules/files/components/navigation-folder.vue

```diff
@@ -113,180 +113,156 @@
 	</div>
 </template>
 
-<script lang="ts">
+<script setup lang="ts">
 import { useI18n } from 'vue-i18n';
-import { defineComponent, PropType, ref } from 'vue';
+import { ref } from 'vue';
 import { useFolders, Folder } from '@/composables/use-folders';
 import api from '@/api';
 import FolderPicker from '@/views/private/components/folder-picker.vue';
 import { useRouter } from 'vue-router';
 import { unexpectedError } from '@/utils/unexpected-error';
 
-export default defineComponent({
-	name: 'NavigationFolder',
-	components: { FolderPicker },
-	props: {
-		folder: {
-			type: Object as PropType<Folder>,
-			required: true,
-		},
-		currentFolder: {
-			type: String,
-			default: null,
-		},
-		clickHandler: {
-			type: Function,
-			default: () => undefined,
-		},
-	},
-	setup(props) {
-		const { t } = useI18n();
-
-		const router = useRouter();
-
-		const { renameActive, renameValue, renameSave, renameSaving } = useRenameFolder();
-		const { moveActive, moveValue, moveSave, moveSaving } = useMoveFolder();
-		const { deleteActive, deleteSave, deleteSaving } = useDeleteFolder();
-
-		const { fetchFolders } = useFolders();
-
-		return {
-			t,
-			renameActive,
-			renameValue,
-			renameSave,
-			renameSaving,
-			moveActive,
-			moveValue,
-			moveSave,
-			moveSaving,
-			deleteActive,
-			deleteSave,
-			deleteSaving,
-		};
-
-		function useRenameFolder() {
-			const renameActive = ref(false);
-			const renameValue = ref(props.folder.name);
-			const renameSaving = ref(false);
-
-			return { renameActive, renameValue, renameSave, renameSaving };
-
-			async function renameSave() {
-				renameSaving.value = true;
-
-				try {
-					await api.patch(`/folders/${props.folder.id}`, {
-						name: renameValue.value,
-					});
-				} catch (err: any) {
-					unexpectedError(err);
-				} finally {
-					renameSaving.value = false;
-					await fetchFolders();
-					renameActive.value = false;
-				}
-			}
+const props = withDefaults(
+	defineProps<{
+		folder: Folder;
+		currentFolder?: string;
+		clickHandler?: () => void;
```

Comment:
`clickHandler` seems to be completely unused (only passed down to a nested `navigation-folder` and never called. Also it is never bound in the one instance that this component is used. Pretty sure it is safe to remove.

Reply:
Is reworked in https://github.com/directus/directus/pull/18320 anyway 👍